### PR TITLE
Panel props

### PR DESCRIPTION
**Pull request by adriana-alter**
Enable the Panel to recalculate its display when it receives new props.  (This was motivated by the UI for registry_admin's [new search feature](https://jira.nypl.org/browse/SIMPLY-1887), whereby the panel containing the search result should automatically open, but there's no problem with merging/publishing by itself rather than waiting for the other repos' search result implementations to be in.)

NOTE: the registry_admin's search feature has changed such that it no longer needs to use this branch, but I think this functionality is still worth having.

### CODE REVIEW

File: src/components/Panel.tsx [hunk 1]
```diff
@@ -42,8 +42,19 @@ export default class Panel extends React.Component<PanelOwnProps, PanelState> {
...
+    this.state = { display: this.calculateDisplay(props) };
+  }
+
+  calculateDisplay(props: PanelOwnProps): string {
+    let display: string;
+    // If the panel isn't collapsible, that overrides everything else.
+    if (!props.collapsible) {
+      display = "";
+    }
+    else {
```
Comment (EdwinGuzman) [inline src/components/Panel.tsx]: This should be on the previous line.

**Comment by EdwinGuzman** [comment]
It doesn't look like this is being used in the Registry Admin. I also don't think this is necessary and complicates what should be a simpler panel component. Since the search results can be more than one library, then it doesn't make sense to have them open.

**Comment by adriana-alter** [comment]
It's not being used in the registry admin, but there are hypothetical scenarios in which a developer using the panel in some other interface might want this functionality, so if the idea is to make these components as reusable/configurable as possible...

**Comment by EdwinGuzman** [comment]
I find it difficult to see when someone would want to add dynamic panels that can be collapsible or open by default. So, I'm reluctant to merge this in. Unless you need it right now for the other PR I'd say just leave this for later when the use case comes up.

**Comment by adriana-alter** [comment]
Okay--will remove from sprint.

**Pull request closed without merging**